src/libraries/System.Console/src/System/IO/StdInReader.cs

```diff
@@ -121,7 +121,7 @@ public int ReadLine(Span<byte> buffer)
                 bytesUsedTotal += bytesUsed;
                 charsUsedTotal += charsUsed;
 
-                if (charsUsed == 0)
+                if (charsUsed < chunk.Span.Length || buffer.IsEmpty)
```

Comment:
Is it not an equivalent of checking if the conversion was complete, which can be done using the `completed` flag?

https://docs.microsoft.com/en-us/dotnet/api/system.text.encoder.convert?view=net-6.0

> The completed output parameter indicates whether all the data in the input buffer was converted and stored in the output buffer. This parameter is set to false if the number of characters specified by the charCount parameter cannot be converted without exceeding the number of bytes specified by the byteCount parameter.

```suggestion
                if (!completed)
```

Reply:
The encoding of the chunk can complete and fill the buffer completely. We still need the `IsEmpty` condition to avoid trying to encode the next chunk.